mautic/mautic
# Ensure $address is not null before passing it to InvalidEmailException

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #13728  



## Description

This PR addresses an issue where the `EmailValidator` class could pass a null value to the `InvalidEmailException` constructor, causing a type error. The changes ensure that the email address passed to the `InvalidEmailException` is always a string.

When the `validate` method of the `EmailValidator` class receives a null value for the `$address` parameter, it leads to a type error in the `InvalidEmailException` constructor. The constructor expects a string but is given null instead, resulting in an error.

Change:
* Modified the `validate` method in `EmailValidator` to handle null or empty email addresses using the null coalescing operator (?:).






---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create new segment
4. Add new contact without e-mail address
5. Add newly created contact to new segment
6. Create new campaign, select contact source, add condition check ("has valid e-mail address")
7. Publish new campaign
8. Execute `mautic:campaign:rebuild`
9. Execute `mautic:campaign:trigger` (should not throw any errors)
